swim/memberlist.go

```diff
@@ -336,11 +336,12 @@ func (m *memberlist) Update(changes []Change) (applied []Change) {
 		// if change is local override, reassert member is alive
 		if member.localOverride(m.node.Address(), change) {
 			m.node.emit(RefuteUpdateEvent{})
+			newIncNo := nowInMillis(m.node.clock)
 			overrideChange := Change{
```

Comment:
To add to it; this code would benefit a lot from a refactor where it does a reincarnation by setting the new incarnation value to a struct describing the state of the node somewhere in memory and from there construct a change that automatically adds the source information. I would even go as far as removing the source information completely from changes, but that will be a different discussion.